boa/src/builtins/value/mod.rs

```diff
@@ -149,6 +151,103 @@ impl Value {
+                let mut new_obj = Object::default();
+                let length = vs.len();
+                for (idx, json) in vs.into_iter().enumerate() {
+                    new_obj.properties.insert(
+                        idx.to_string(),
+                        Property::default()
+                            .value(Self::from_json(json, interpreter))
+                            .writable(true)
+                            .configurable(true),
+                    );
+                }
+                new_obj.properties.insert(
+                    "length".to_string(),
+                    Property::default().value(Self::from(length)),
+                );
+                Self::object(new_obj)
+            }
+            JSONValue::Object(obj) => {
+                let new_obj = Value::new_object(Some(&interpreter.realm.global_obj));
+                for (key, json) in obj.into_iter() {
+                    let value = Self::from_json(json, interpreter);
+                    new_obj.set_property(
+                        key,
+                        Property::default()
+                            .value(value)
+                            .writable(true)
+                            .configurable(true),
+                    );
+                }
+                new_obj
+            }
+            JSONValue::Null => Self::null(),
+        }
+    }
+
+    /// Conversts the `Value` to `JSON`.
+    pub fn to_json(&self, interpreter: &mut Interpreter) -> Result<JSONValue, Value> {
+        match *self.data() {
+            ValueData::Null => Ok(JSONValue::Null),
+            ValueData::Boolean(b) => Ok(JSONValue::Bool(b)),
+            ValueData::Object(ref obj) => {
+                if obj.borrow().kind == ObjectKind::Array {
+                    let mut arr: Vec<JSONValue> = Vec::new();
+                    obj.borrow().properties.keys().for_each(|k| {
+                        if k != "length" {
+                            let value = self.get_field(k.to_string());
+                            if value.is_undefined() || value.is_function() {
+                                arr.push(JSONValue::Null);
+                            } else {
+                                arr.push(
+                                    self.get_field(k.to_string()).to_json(interpreter).unwrap(),
+                                );
+                            }
+                        }
+                    });
+                    Ok(JSONValue::Array(arr))
+                } else {
+                    let mut new_obj = Map::new();
+                    obj.borrow().properties.keys().for_each(|k| {
+                        let key = k.clone();
+                        let value = self.get_field(k.to_string());
+                        if !value.is_undefined() && !value.is_function() {
+                            // TODO: Handle the errors
+                            new_obj.insert(key, value.to_json(interpreter).unwrap());
```

Comment:
Another place where `?` should be used instead of `unwrap()`.

Reply:
Same here about being inside a for_each.

Reply:
I would change the `for_each()` for a `for k in obj.borrow().properties.keys()` here too.

Reply:
:heavy_check_mark: